source/common/http/conn_manager_impl.cc

```diff
@@ -1058,6 +1062,13 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
     connection_manager_.drain_state_ = DrainState::Closing;
   }
 
+  if (connection_manager_.drain_state_ == DrainState::NotDraining &&
+      connection_manager_.overload_disable_keepalive_ref_ == Server::OverloadActionState::Active) {
+    ENVOY_STREAM_LOG(debug, "closing connection due to envoy overload", *this);
```

Comment:
Can you clarify the log message to indicate that its related to keepalive?

Reply:
done